torch/utils/data/dataloader.py

```diff
@@ -213,6 +215,8 @@ class DataLoader(Generic[_T_co]):
     .. warning:: See :ref:`reproducibility`, and :ref:`dataloader-workers-random-seed`, and
                  :ref:`data-loading-randomness` notes for random seed related questions.
 
+    .. warning:: Setting `allow_out_of_order` to `True` can harm reproducibility.
```

Comment:
Another warning to call out: in the case of imbalanced data it may lead to skewed data distribution being fed to the trainer, where eg batches with small data/faster-to-pre-proc may introduce unwanted data-skew.